packages/vm/src/runBlock.ts

```diff
@@ -42,6 +42,7 @@ const DAORefundContract = DAOConfig.DAORefundContract
 export async function runBlock(this: VM, opts: RunBlockOpts): Promise<RunBlockResult> {
   const state = this.eei
   const { root } = opts
+  const clearCache = opts.clearCache ?? true
```

Comment:
I think what we should do here is:

1. Check current state root
2. Check parent header state root
3. If this is equal, then do not clear cache
4. If it is non-equal then clear cache
5. Of course if `opts.clearCache` is defined this overrides above logic.

Reply:
Hmm, I am not seeing how we can do this without going out of the scope of this method. I think storing the block from the previous `runBlock()` execution and then comparing to that is definitely not the right approach respectively happening "at the wrong place", this would rather be something for the client (if at all).

For VM one can have all sorts of setups (execute `runBlock()`, manually set to an own state root, run another block, whatever). So this narrows things down to much.

Let me know if I understand something wrong about your suggestion though.

Reply:
I have no strong opinion either, current implementation is fine!